src/nginx/util.h

```diff
@@ -65,6 +65,11 @@ inline bool ngx_string_equal(const ngx_str_t &str1, const ngx_str_t &str2) {
   return str1.len == str2.len && !ngx_strncmp(str1.data, str2.data, str1.len);
 }
 
+// Check an ngx_str_t against a prefix ngx_str_t
+inline bool ngx_string_startswith(const ngx_str_t &subject, const ngx_str_t &prefix) {
+  return subject.len > prefix.len && ngx_strstr(subject.data, prefix.data) == 0;
```

Comment:
added a util method `ngx_string_startswith`. to opt-in to upstream TLS when communicating over gRPC, simply prefix the endpoint with `grpcs://`.